app/src/main/java/hm/orz/chaos114/android/tumekyouen/util/EncryptionUtil.java

```diff
@@ -8,108 +11,102 @@
-			key = new SecretKeySpec(keyBytes, "AES");
-		}
-	}
+            // キーを復元
+            byte[] keyBytes = Base64.decode(keyStr, Base64.URL_SAFE
+                    | Base64.NO_WRAP);
+            key = new SecretKeySpec(keyBytes, "AES");
+        }
+    }
 
-	/**
-	 * 暗号化した文字列を返却する。
-	 * 
-	 * @param input 入力文字列
-	 * @return 暗号化した文字列
-	 */
-	public String encrypt(String input) {
-		if (input == null) {
-			return null;
-		}
+    /**
+     * 暗号化キーを生成する。
+     *
+     * @return 暗号化キー
+     */
+    private static Key generateKey() {
+        try {
+            KeyGenerator generator = KeyGenerator.getInstance("AES");
+            SecureRandom random = SecureRandom.getInstance("SHA1PRNG");
+            generator.init(ENCRYPT_KEY_LENGTH, random);
+            return generator.generateKey();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
 
-		try {
-			Cipher cipher = Cipher.getInstance("AES");
-			cipher.init(Cipher.ENCRYPT_MODE, key);
-			byte[] result = cipher.doFinal(input.getBytes());
-			return Base64.encodeToString(result, Base64.URL_SAFE
-					| Base64.NO_WRAP);
-		} catch (GeneralSecurityException e) {
-			throw new RuntimeException(e);
-		}
-	}
+    /**
+     * 暗号化した文字列を返却する。
+     *
+     * @param input 入力文字列
+     * @return 暗号化した文字列
+     */
+    public String encrypt(String input) {
+        if (input == null) {
+            return null;
+        }
 
-	/**
-	 * 復号化した文字列を返却する。
-	 * 
-	 * @param input 入力文字列
-	 * @return 復号化した文字列
-	 */
-	public String decrypt(String input) {
-		if (input == null) {
-			return null;
-		}
+        try {
+            Cipher cipher = Cipher.getInstance("AES");
+            cipher.init(Cipher.ENCRYPT_MODE, key);
+            byte[] result = cipher.doFinal(input.getBytes());
+            return Base64.encodeToString(result, Base64.URL_SAFE
+                    | Base64.NO_WRAP);
+        } catch (GeneralSecurityException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
-		try {
-			Cipher cipher = Cipher.getInstance("AES");
-			cipher.init(Cipher.DECRYPT_MODE, key);
-			byte[] result = cipher.doFinal(Base64.decode(input, Base64.URL_SAFE
-					| Base64.NO_WRAP));
-			return new String(result);
-		} catch (GeneralSecurityException e) {
-			throw new RuntimeException(e);
-		}
-	}
+    /**
+     * 復号化した文字列を返却する。
+     *
+     * @param input 入力文字列
+     * @return 復号化した文字列
+     */
+    public String decrypt(String input) {
+        if (input == null) {
+            return null;
+        }
 
-	/**
-	 * 暗号化キーを生成する。
-	 * 
-	 * @return 暗号化キー
-	 */
-	private static Key generateKey() {
-		try {
-			KeyGenerator generator = KeyGenerator.getInstance("AES");
-			SecureRandom random = SecureRandom.getInstance("SHA1PRNG");
-			generator.init(ENCRYPT_KEY_LENGTH, random);
-			return generator.generateKey();
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
-	}
+        try {
+            Cipher cipher = Cipher.getInstance("AES");
```

Comment:
WARNING: [GetInstance] `Cipher.getInstance` should not be called without setting the encryption mode and padding
 `Cipher#getInstance` should not be called with ECB as the cipher mode or without setting the cipher mode because the default mode on android is ECB, which is insecure.